### DISCUSSION TITLE
Question about QES

### DISCUSSION THREAD

**Discussion by tlodderstedt** [eu-digital-identity-wallet/eudi-wallet-reference-implementation-roadmap#49] [Q&A]
I have some questions re https://github.com/eu-digital-identity-wallet/eudi-wallet-reference-implementation-roadmap as we are currently developing our product specification for the DE EUDIW implementation and would like to align with your thoughts. 

How will the setup/provisioning process look alike? I'm asking since according to the regulation, the EUDIW shall manage the siging certificate. So I would assume there will be some issuance process in place to install the certificate into the wallet as a basis for eQES.
What party determines the QSCD used for signature creation?
How will the signature creation be authenticated/authorized?

**Comment by pinamiranda** [comment]
Hi @tlodderstedt , Thank you for your input. 
Could you please clarify what you mean by “according to the regulation, the EUDIW shall manage the signing certificate”? Are you referring to Article 5a(4)(e)?
Additionally, could you confirm if “eQES” was a typo and you actually meant “rQES”?

**Comment by tlodderstedt** [comment]
@pinamiranda I'm referring to article 5a (5), in particular to the following text: "
5. European Digital Identity Wallets shall, in particular:
(a) support common protocols and interfaces:
(i) for issuance of ... or qualified and non-qualified
certificates to the European Digital Identity Wallet;"

yes, I meant "rQES" as in "Remote QES".

**Comment by pinamiranda** [comment]
@tlodderstedt Thank you.

In the current implementation of the rQES, the QTSP server issues a qualified certificate for the user if no existing certificate is found.

This process occurs after the user authenticates with the EUDIW system within the QTSP. Once the user is authenticated and requests the list of available certificates, if no certificates are found, a new qualified certificate is issued. This flow is illustrated in the sequence diagram provided at the following link: https://github.com/eu-digital-identity-wallet/eudi-srv-web-walletdriven-signer-qtsp-java#credentials-listing

To retrieve the generated certificate, the '/credentials/list' endpoint, as specified in the CSC API V2.0 Specification, can be invoked.

**Reply by tlodderstedt** [reply]
thanks

**Reply by sander** [reply]
As I see it, this is indeed a valid way in which the QTSP could implement issuance of long-term certificates. Some other ways I see with the current CSC v2.0 and v2.1 specs:

- just-in-time issuance as-needed upon `credentials/list` as implemented here; downside is that typically issuance takes longer than the expected duration of a `credentials/list` request;
- just-in-time issuance as-needed during the first OAuth 2.0 authorisation process for `credential` scope with CSCv2 `signatureQualifier = eu_eidas_qes`;
- proactive issuance during initial enrolment with the QTSP, implemented using an OAuth 2.0 authorisation process for CSCv2 `service` scope;
- proactive issuance during initial enrolment with the QTSP, implemented using an OpenID4VCI credential request by the Wallet;
- proactive issuance during initial enrolment with the QTSP, implemented using a new API.

Note that for most options, the EUDIW does not need to be aware of the mechanism selected by the QTSP. My personal favourite for now is upon `service` authorisation, expiring access and refresh tokens as soon as the long-term certificate has expired.
